src/docfx/Errors.cs

```diff
@@ -4,174 +4,390 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 
 namespace Microsoft.Docs.Build
 {
     internal static class Errors
     {
+        /// <summary>
+        /// Defined same redirection entry in both config's redirections and redirectionsWithoutId.
+        /// </summary>
         public static Error RedirectionConflict(string redirectFrom)
             => new Error(ErrorLevel.Error, "redirection-conflict", $"The '{redirectFrom}' appears twice or more in the redirection mappings");
 
+        /// <summary>
+        /// Redirection entry isn't a conceptual article(*.{md,json,yml}).
+        /// </summary>
         public static Error InvalidRedirection(string path, ContentType contentType)
             => new Error(ErrorLevel.Error, "invalid-redirection", $"The '{path}' shouldn't belong to redirections since it's a {contentType}");
 
+        /// <summary>
+        /// Used invalid glob pattern in configuration.
+        /// Examples:
+        ///   - in build scope include/exclude files
+        ///   - in file metadata glob
+        /// </summary>
         public static Error InvalidGlobPattern(string pattern, Exception ex)
             => new Error(ErrorLevel.Error, "invalid-glob-pattern", $"The glob pattern '{pattern}' is invalid: {ex.Message}");
 
+        /// <summary>
+        /// Docfx.yml/docfx.json doesn't exist at the repo root.
+        /// Examples:
+        ///   - non-loc build, docfx.yml/docfx.json doesn't exist at the repo root
+        ///   - loc build, docfx.yml/docfx.json doesn't exist at the neither loc repo root
+        ///     nor source repo root
+        /// </summary>
         public static Error ConfigNotFound(string docsetPath)
             => new Error(ErrorLevel.Error, "config-not-found", $"Cannot find 'docfx.yml/docfx.json' at '{docsetPath}'");
 
+        /// <summary>
+        /// Two files include each other.
```

Comment:
not only markdown inclusion, xref circular reference is using this error code as well